src/core/config.c

```diff
@@ -0,0 +1,137 @@
+#include "config.h"
+#include "xdpw.h"
+#include "logger.h"
+
+#include <dictionary.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include "iniparser.h"
+
+void print_config(enum LOGLEVEL loglevel, struct xdpw_config *config) {
+	logprint(loglevel, "config: Configfile %s",config->conf.configfile);
+	logprint(loglevel, "config: Loglevel %d",config->logger_conf.loglevel);
+	logprint(loglevel, "config: Outputname  %s",config->screencast_conf.output_name);
+}
+
+// NOTE: calling destroy_config won't prepare the config to be read again from config file
+// with init_config since to pointers and other values won't be reset to NULL, or 0
+void destroy_config(struct xdpw_config *config) {
+	logprint(DEBUG, "config: destroying config");
+
+	free(&config->conf.configfile);
+
+	// screencast
+	free(&config->screencast_conf.output_name);
+}
+
+void getstring_from_conffile(dictionary *d, char *key, char **dest, char *def) {
+	if (*dest != NULL) {
+		return;
+	}
+	const char *c = iniparser_getstring(d, key, def);
+	if (c != NULL) {
+		// Allow keys without value as default
+		if (strcmp(c, "") != 0) {
+			*dest = strdup(c);
+		} else {
+			*dest = strdup(def);
+		}
+	}
+}
+
+void getenum_from_conffile(dictionary *d, char *key, int *dest, const char *array[] , int num_array, int def) {
```

Comment:
This sounds like unnecessary complexity here. If we want to keep the log level config option, why not have a function that converts a string to a log level?

Reply:
We also need this later for the chooser type in #59 so we need a function to convert a string to an enum via a string array. So we could read the string from the config file and convert it with a seperate function, or do both at once.

Reply:
I don't see a real difference between an array and a function that parses a string. At least, the function ensures the array items aren't mixed up.

Reply:
you mean the problem, that someone could change the order of the array, which indexes should correspond to the enum values?

This would mean, we need one function for every enum?

Reply:
> you mean the problem, that someone could change the order of the array, which indexes should correspond to the enum values?

Yeah. Or just add a new entry out-of-order.

> This would mean, we need one function for every enum?

One function, one array, no real difference. Better keep things simple.

Reply:
sure. i guess i should not replace the array with a function pointer to the new compare functions and keep the nice syntax?

Reply:
Just keep it simple. No function pointer, just an intermediate `char *` variable that we feed to the function parsing the string.

Reply:
so sad. ok no problem.

Reply:
Note: we totally relied on this array before

Reply:
My main beef is the generic function for enums.